### ISSUE TITLE
Simple code with varargs fails with exception

### ISSUE THREAD

**Issue by dours** [cqfn/eo#414]
The test:
```
[] > varargs-fail
  [args...] > f
    1 > a
    2 > @
  (f 1 2 3).eq 2 > @
```
leads to "You can't overwrite X'

**Comment by dours** [comment]
Looks very similar to #384

**Comment by yegor256** [comment]
@dours looks like a bug, thanks! let me investigate

**Comment by dours** [comment]
Just moving an .eo file is not enough, so I currently expect help of @PetrB09 here

**Comment by baudoliver7** [comment]
@yegor256 ready to take this.

**Comment by yegor256** [comment]
@baudoliver7 go ahead!

**Comment by baudoliver7** [comment]
@yegor256 command line `mvn clean compile` under `sandbox/hse` doesn't work for me.
I got this :
```cmd
[INFO] Scanning for projects...
[INFO] 
[INFO] ---------------------< org.eolang:quick-start-hse >---------------------
[INFO] Building quick-start-hse 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.301 s
[INFO] Finished at: 2021-12-22T11:03:38Z
[INFO] ------------------------------------------------------------------------
[ERROR] Could not find goal 'compile' in plugin org.eolang:eo-maven-plugin:0.16.1 among available goals assemble, copy, discover-foreign, extend, help, optimize, parse, place, pull, register, resolve, transpile, unplace, unspile -> [Help 1]
[ERROR] 
...
```
What am I missing ? (`mvn clean install` works well under `eo` project).

**Comment by yegor256** [comment]
@baudoliver7 how is this related to the bug reported above?

**Comment by baudoliver7** [comment]
@yegor256 I just want to run the code sample to note the bug.

**Comment by yegor256** [comment]
@baudoliver7 run it using `sandbox/canonical`

**Comment by baudoliver7** [comment]
@yegor256 Yes, I'm already using it. Thx :)

**Comment by baudoliver7** [comment]
@dours @yegor256
It's not because of `varags`. The code below also fails with the same message when assigning value to `a` :
```eo
[] > varargs-fail
  [] > f
    1 > a
    2 > @
  (f 1 2 3).eq 2 > @
```

**Comment by 0pdd** [comment]
@dours 2 puzzles [#534](https://github.com/cqfn/eo/issues/534), [#535](https://github.com/cqfn/eo/issues/535) are still not solved.

**Comment by 0pdd** [comment]
@dours 2 puzzles [#622](https://github.com/objectionary/eo/issues/622), [#623](https://github.com/objectionary/eo/issues/623) are still not solved.

**Comment by 0pdd** [comment]
@dours the puzzle [#622](https://github.com/objectionary/eo/issues/622) is still not solved; solved: [#623](https://github.com/objectionary/eo/issues/623).

**Comment by 0pdd** [comment]
@dours all 2 puzzles are solved here: [#622](https://github.com/objectionary/eo/issues/622), [#623](https://github.com/objectionary/eo/issues/623).

**Comment by yegor256** [comment]
@dours should work now, try version 0.23.5
